iOSDFULibrary/Classes/Utilities/DFUPackage/Manifest/ManifestFirmwareInfo.swift

```diff
@@ -30,20 +30,30 @@
 
 import Foundation
 
-class ManifestFirmwareInfo: NSObject {
-    var binFile: String? = nil
-    var datFile: String? = nil
+// MARK: - ManifestFirmware
+
+protocol ManifestFirmware {
+    
+    var binFile: String? { get }
+    var datFile: String? { get }
+}
+
+extension ManifestFirmware {
     
     var valid: Bool {
-        return binFile != nil // && datFile != nil The init packet was not required before SDK 7.1
```

Comment:
By changing this we break compatibility with Swift 4.2 and 5.0. This was introduced in Swift 5.1.
Have a look at the podspec: https://github.com/NordicSemiconductor/IOS-DFU-Library/blob/d9aa014c363f1bd23bb1fbc2f756c9607bbd8870/iOSDFULibrary.podspec#L14
We may also just change the podspec.